packages/operation/src/evm/deploy/providers.ts

```diff
@@ -0,0 +1,45 @@
+import { ethers } from 'ethers';
+import { deriveProviderId } from '../utils';
+import { DeployState as State } from '../../types';
+
+export async function createProviders(state: State): Promise<State> {
+  for (const apiProviderName of Object.keys(state.apiProvidersByName)) {
+    const apiProvider = state.apiProvidersByName[apiProviderName];
+
+    // Ensure that the API provider address has enough ETH to create the onchain provider
+    await state.deployer.sendTransaction({ to: apiProvider.address, value: ethers.utils.parseEther('5') });
+
+    await state.contracts
+      .Airnode!.connect(apiProvider.signer)
+      .createProvider(apiProvider.address, apiProvider.xpub, { value: ethers.utils.parseEther('1') });
+  }
+  return state;
+}
+
+export async function authorizeEndpoints(state: State): Promise<State> {
+  const { Airnode } = state.contracts;
+
+  for (const providerName of Object.keys(state.apiProvidersByName)) {
+    const configApiProvider = state.config.apiProviders[providerName];
+    const apiProvider = state.apiProvidersByName[providerName];
+    const providerId = deriveProviderId(apiProvider.address);
+
+    for (const endpointName of Object.keys(configApiProvider.endpoints)) {
+      const configEndpoint = configApiProvider.endpoints[endpointName];
+
+      const { keccak256, defaultAbiCoder } = ethers.utils;
+      const endpointId = keccak256(defaultAbiCoder.encode(['string'], [endpointName]));
```

Comment:
It's `${oisTitle}/${endpointName}`, I assume you're only using the endpoint name for simplicity (which is okay)

Reply:
Nice catch - I forgot about this. I'd rather keep it the same just to avoid confusion.